src/Biogeochemistry.jl

```diff
@@ -0,0 +1,138 @@
+module Biogeochemistry
+
+using Oceananigans.Grids: Center, xnode, ynode, znode
+using Oceananigans.Advection: div_Uc, CenteredSecondOrder
+using Oceananigans.Utils: tupleit
+using Oceananigans.Fields: Field
+using Oceananigans.Architectures: device, architecture
+using Oceananigans.Utils: launch!
+using KernelAbstractions: @kernel
+
+import Oceananigans.Fields: location, CenterField
+import Oceananigans.Forcings: regularize_forcing, maybe_constant_field
+
+#####
+##### Generic fallbacks for biogeochemistry
+#####
+
+"""
+Update tendencies.
+
+Called at the end of calculate_tendencies!
+"""
+update_tendencies!(bgc, model) = nothing
+
+"""
+Update tracer tendencies.
+
+Called at the end of calculate_tendencies!
+"""
+update_biogeochemical_state!(bgc, model) = nothing
+
+@inline biogeochemical_drift_velocity(bgc, val_tracer_name) = nothing
+@inline biogeochemical_advection_scheme(bgc, val_tracer_name) = nothing
+@inline biogeochemical_auxiliary_fieilds(bgc) = NamedTuple()
+
+#####
+##### Default (discrete form) biogeochemical source
+#####
+
```

Comment:
probably it is good to describe in a docstring what `AbstractBioGeochemistry` is and what is required of a type that subtypes it (the same of `AbstractContinuousFormBiogechimstry`)

Reply:
Not sure if its exactly what you wanted but had a go at this now

Reply:
looks good

Reply:
I think we should leave this here... but note that I don't think many other abstract types have detailed docstrings. The pitfall is that the docstring has to be maintained if the interface changes.